datatypes/mcs_int128.h

```diff
@@ -203,6 +180,36 @@ class TSInt128
       return s128Value == static_cast<int128_t>(x);
     }
 
+    inline operator double() const
+    {
+      return toDouble();
+    }
+
+    inline long double toDouble() const
+    {
+      return static_cast<double>(s128Value);
```

Comment:
I think this conversion can overflow, this should be `return static_cast<double>(TFloat128(s128Value))`.